.github/workflows/prepare-release.yml

```diff
@@ -0,0 +1,154 @@
+          if [ "${BLOCKERS}" != "" ]; then
+              echo "Release blockers in ${REPO} repo: ${BLOCKERS}"
+              exit 1
+          fi
+      # Make sure that there are no open issues with release:blocker label in Contrib. The release has to be delayed until they are resolved.
+      - name: Check blockers in contrib
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          REPO: open-telemetry/opentelemetry-collector-contrib
+        run: |
+          BLOCKERS=$( gh issue list --search "label:release:blocker" --json url --jq '.[].url' --repo ${REPO} )
+          if [ "${BLOCKERS}" != "" ]; then
+              echo "Release blockers in ${REPO} repo: ${BLOCKERS}"
+              exit 1
+          fi
+      # Make sure the current main branch build successfully passes (Core).
+      - name: Check build status in core
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          REPO: open-telemetry/opentelemetry-collector
+        run: |
+          RESULT=$(gh run list --branch main --json status --jq '[.[] | select(.status != "queued" and .status != "in_progress")][0].status' --workflow build-and-test --repo ${REPO} )
+          if [ "${RESULT}" != "completed" ]; then
+              echo "Build status in ${REPO} is not completed: ${RESULT}"
+              echo "$(gh run list --branch main --json status,url --jq '[.[] | select(.status != "queued" and .status != "in_progress")][0].url' --workflow build-and-test --repo ${REPO} )"
+              exit 1
+          fi
+      # Make sure the current main branch build successfully passes (Contrib).
+      - name: Check build status in contrib
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          REPO: open-telemetry/opentelemetry-collector-contrib
+        run: |
+          RESULT=$(gh run list --branch main --json status --jq '[.[] | select(.status != "queued" and .status != "in_progress")][0].status' --workflow build-and-test --repo ${REPO} )
+          if [ "${RESULT}" != "completed" ]; then
+              echo "Build status in ${REPO} is not completed: ${RESULT}"
+              echo "$(gh run list --branch main --json status,url --jq '[.[] | select(.status != "queued" and .status != "in_progress")][0].url' --workflow build-and-test --repo ${REPO} )"
+              exit 1
+          fi
+      # Update Contrib to use the latest in development version of Core. Run make update-otel in
+      # Contrib root directory and if it results in any changes, submit a PR. Open this PR as draft.
+      # This is to ensure that the latest core does not break contrib in any way. We’ll update it
+      # once more to the final release number later.
+      - name: Make update-otel in contrib
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          REPO: open-telemetry/opentelemetry-collector-contrib
+        run: "echo TODO" # TODO: This may require a webhook in contrib
```

Comment:
Or perhaps an approver's PAT as a secret?

Reply:
It's possible, I've removed that step for now as it's still documented in the release doc as manual